hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java

```diff
@@ -122,6 +129,32 @@ public CallOutcome call() {
         }
     }
 
+    private HederaEvmTransaction safeCreateHevmTransaction() {
+        try {
+            return hevmTransactionFactory.fromHapiTransaction(context.body());
+        } catch (HandleException e) {
+            if (context.body().hasContractCall()) {
+                // Return a HederaEvmTransaction that represents the error in order to charge fees to the sender
+                return hevmTransactionFactory.fromContractCallException(
+                        context.body().contractCallOrThrow(), e);
+            }
+            throw e;
+        }
+    }
+
+    private CallOutcome chargeFeesAndReturnOutcome(HederaEvmTransaction hevmTransaction) {
+        // if there was an exception while creating the HederaEvmTransaction and the transaction is a ContractCall
```

Comment:
Q: should this also be utilized by `ContractCreate` and `EthereumTransaction`

Reply:
The hole we are covering with this pr is specific to `ContractCall` as these calls uniquely do not charge other hbar denominated fees.  We could charge for the `ContractCreate` and `EthereumTransaction` but as far as I know there hasn't been a decision made to charge such fees.